hs_core/management/commands/ingest_irods_files.py

```diff
@@ -0,0 +1,190 @@
+        else:
+            return __ingest_irods_directory(self,
+                                            self.file_path,
+                                            logger,
+                                            stop_on_error=False,
+                                            echo_errors=True,
+                                            log_errors=False,
+                                            return_errors=False)
+
+    return errors, ecount
+
+
+def __ingest_irods_directory(self,
+                             dir,
+                             logger,
+                             stop_on_error=False,
+                             log_errors=True,
+                             echo_errors=False,
+                             return_errors=False):
+    """
+    list a directory and check files there for conformance with django ResourceFiles
+
+    :param stop_on_error: whether to raise a ValidationError exception on first error
+    :param log_errors: whether to log errors to Django log
+    :param echo_errors: whether to print errors on stdout
+    :param return_errors: whether to collect errors in an array and return them.
+
+    """
+    errors = []
+    ecount = 0
+    istorage = self.get_irods_storage()
+    try:
+        listing = istorage.listdir(dir)
+        for fname in listing[1]:  # files
+            fullpath = os.path.join(dir, fname)
+            found = False
+            for f in self.files.all():
+                if f.storage_path == fullpath:
+                    found = True
+                    break
+            if not found:
+                ecount += 1
+                msg = "ingest_irods_files: file {} in iRODs does not exist in Django (INGESTING)"\
+                    .format(fullpath)
+                if echo_errors:
+                    print(msg)
+                if log_errors:
+                    logger.error(msg)
+                if return_errors:
+                    errors.append(msg)
+                if stop_on_error:
+                    raise ValidationError(msg)
+                # TODO: only works properly for generic and composite resources!
+                link_irods_file_to_django(self, fullpath)
+
+        for dname in listing[0]:  # directories
+            error2, ecount2 = __ingest_irods_directory(self,
+                                                       os.path.join(dir, dname),
+                                                       logger,
+                                                       stop_on_error=stop_on_error,
+                                                       echo_errors=echo_errors,
+                                                       log_errors=log_errors,
+                                                       return_errors=return_errors)
+            errors.extend(error2)
+            ecount += ecount2
+
+    except SessionException:
+        pass  # not an error not to have a file directory.
+        # Non-existence of files is checked elsewhere.
```

Comment:
It'd be more informative if the SessionException errors are logged/printed before calling ```pass``` just in case.